### PR TITLE
Implemented SIPickleDataInterfaces

### PR DESCRIPTION
**Pull request by alejoe91**
This PR adds the `SIPickleRecordingExtractorInterface` and `SIPickleSortingExtractorInterface`. 

This allows to process the data in the SI pipeline (e.g. synchronizing the recording and curating the sorting). Using the `se.load_extractor_from_pickle()` function, the converter is now capable of retrieving the state of the processed extractors and save it to NWB.

@bendichter @CodyCBakerPhD @luiztauffer what do you guys think?

### CODE REVIEW

**Comment by bendichter** [comment]
Would it make more sense to inherit from https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/baserecordingextractorinterface.py?

**Comment by alejoe91** [comment]
I looked into it, but not really because the pickle loader doesn't really have an `RX`, and the input is only the `pkl_file`

**Comment by alejoe91** [comment]
It makes sense though to have a `BaseSIPickleDataInterface`, since the `get_metadata_schema` is the same.

File: make_env.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: nwb_conversion_env
+name: catalyst
```
Comment (alejoe91) [inline make_env.yml:1]:
```suggestion
name: nwb_conversion_env
```

File: setup.py [hunk 2]
```diff
@@ -25,9 +25,9 @@
     include_package_data=True,
     install_requires=[
         'pynwb', 'tqdm', 'natsort', 'numpy', 'scipy', 'bs4',
-        'pandas', 'jupyter', 'matplotlib', 'h5py', 'pyyaml', 'jupyter-client',
+        'pandas', 'jupyter', 'matplotlib', 'h5py', 'pyyaml',
```
Comment (alejoe91) [inline setup.py:28]: @bendichter I changed the order here because I was getting some conflicts otherwise

**Comment by CodyCBakerPhD** [comment]
I'll check through this, add that magic function we talked about, and refactor stuff within the next 24 hours so we can get this through and cleared off the ToDo stack.

File: setup.py [hunk 3]
```diff
@@ -25,9 +25,9 @@
...
         'spikeextractors', 'spikesorters', 'spiketoolkit',
-        'PySide2', 'nwbwidgets', 'psutil', 'voila'
+        'PySide2', 'psutil', 'voila', 'jupyter-client'
```
Comment (bendichter) [inline setup.py:30]:
```suggestion
```

**Pull request merged**